### ISSUE TITLE
Twitch integration broken?

### ISSUE THREAD

**Issue by onli** [stefansundin/rssbox#42]
To me it seems like the Twitch integration is not working anymore. The Readme points to https://www.twitch.tv/settings/connections, but it seems the application management has moved to https://dev.twitch.tv/console/apps/. It's possible to create an app there and you also get a `clientid`, but if you set that clientid Twitch complains about the Oauth token:

     2020-06-25T15:38:46.102522+00:00 app[web.1]: 2020-06-25 15:38:46 - TwitchError - 401: {"error":"Unauthorized","status":401,"message":"OAuth token is missing"}:

That's when trying to get a feed for a user.

Did Twitch break the rss box API access, or am I misunderstanding what data needs to be provided?

**Comment by stefansundin** [comment]
Hmm. It appears that it is because of this: https://discuss.dev.twitch.tv/t/requiring-oauth-for-helix-twitch-api-endpoints/23916

For a workaround, use the client-id that this Twitch extension is using (it is in the URL): https://dashboard.twitch.tv/extensions/0qr7fa6llzn4txgnfgb8ipeksd5v24

I will update the code later to make this work with your own Client-ID at some point. Thanks for letting me know!

**Comment by stefansundin** [comment]
Oh, looks like they broke embeds too.. is this the warning signs that Twitch is starting to lock down their platform ala Instagram? Boo!

![twitch-embed](https://user-images.githubusercontent.com/1991151/85909095-ff244580-b7cc-11ea-87ce-c97d21c73092.png)

In the console:
> Twitch embed error message:
> [NoParent] parent query string value was not specified
> Docs: https://dev.twitch.tv/docs/embed/video-and-clips/

So I need to start telling them the website that is embedding with `&parent=example.com`? What total and utter garbage.

**Comment by onli** [comment]
That parent parameter is one of the most stupid things I ever saw on the internet. Completely defeats the purpose of embeds - and that in a world where youtube embeds do not need this.

I never saw the feed rssbox would produce for twitch. It would include those embeds?

**Comment by stefansundin** [comment]
Yes, so if your RSS reader supports iframes, you can watch it right there. You also get the embeds if you subscribe to other services, e.g. Twitter and the person tweets a link to a Twitch video or channel.

Apparently the claim is that they need it for CSP headers.. but I don't see why they can't just allow whatever Origin that the web browser sends.. Hmm.. I'll see what I think after I have time to look at it more.

**Comment by stefansundin** [comment]
I have worked around the embedding issue now.

**Comment by stefansundin** [comment]
For those hosting themselves, you now have to use:

```
TWITCHTOKEN_CLIENT_ID=b31o4btkqth5bzbvr9ub2ovr79umhh
```

**Comment by amblerkr** [comment]
> This endpoint should not be used by a robot. RSS Box is open source so you should instead reimplement the thing you need in your own application.

I created a new app and pasted the client ID, but it doesn't work. The client ID you told me does not work either. Is it completely dead?
